frontend/src/Context/GroupsPageContext.js

```diff
@@ -14,122 +14,132 @@ const GroupsPageProvider = ({ children }) => {
+    } catch (error) {
+      console.error("Error fetching groups", error);
+      setError("We were unable to fetch groups at this moment");
     }
   };
 
   const handleSaveGroup = (group) => {
-    axiosInstance.post(`/api/groups`, group)
-      .then(response => {
+    axiosInstance
+      .post(`/api/groups`, group)
+      .then((response) => {
         const newGroup = response.data;
-        setGroups(prevGroups => ({
+        setGroups((prevGroups) => ({
           ...prevGroups,
-          [newGroup.name]: newGroup.members || []
+          [newGroup.name]: newGroup.members || [],
         }));
-        setGroupIDS(prevIDs => [...prevIDs, newGroup._id]);
+        setGroupIDS((prevIDs) => [...prevIDs, newGroup._id]);
         setShowModal(false);
       })
       .catch((error) => console.error("Error creating a new group", error));
-  }
+  };
 
   const addMember = (groupName, userId) => {
-    axiosInstance.get(`/api/groups`)
-      .then(response => {
-        const group = response.data.find(grp => grp.name === groupName);
+    axiosInstance
+      .get(`/api/groups`)
+      .then((response) => {
+        const group = response.data.find((grp) => grp.name === groupName);
         if (group) {
           const groupId = group._id;
-          axiosInstance.put(`/api/groups/${groupId}/add-member`, { userId })
+          axiosInstance
+            .put(`/api/groups/${groupId}/add-member`, { userId })
             .then((response) => {
-              setGroupIDS(prevIDs => [...prevIDs, response.data._id])
+              setGroupIDS((prevIDs) => [...prevIDs, response.data._id]);
               fetchGroups();
               setShowMemberModal(false);
             })
-            .catch(error => console.error('Error adding member to group', error));
+            .catch((error) =>
+              console.error("Error adding member to group", error)
+            );
         } else {
-          console.error('No such group');
+          console.error("No such group");
         }
       })
-      .catch(error => console.error('Error fetching groups', error));    
-  }
+      .catch((error) => console.error("Error fetching groups", error));
+  };
   const handleAddGroup = () => {
     setCurrentGroup(null);
     setShowModal(true);
-  }
+  };
 
   const handleAddMember = (group) => {
     setSelectedGroup(group);
     setShowMemberModal(true);
-  }
+  };
 
   const getGroupIDs = () => {
-    axiosInstance.get(`/api/groups`)
-      .then(response => {
-        const ids = response.data.map(grp => grp._id);
+    axiosInstance
+      .get(`/api/groups`)
+      .then((response) => {
+        const ids = response.data.map((grp) => grp._id);
         setGroupIDS(ids);
       })
-      .catch(error => console.error('Error fetching Group IDS', error));
+      .catch((error) => console.error("Error fetching Group IDS", error));
   };
 
   useEffect(() => {
     const fetchMemberRequests = async () => {
```

Comment:
code design suggestion: declare the function outside of the useEffect and use it in the useeffect. 
Although this is fine